src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -85,36 +80,86 @@ DatabaseShardImp::init()
             "Already initialized";
         return false;
     }
-    if (ledgersPerShard_ == 0 || ledgersPerShard_ % 256 != 0)
+
+    auto fail = [&](std::string const& msg)
     {
         JLOG(j_.error()) <<
-            "ledgers_per_shard must be a multiple of 256";
+            "[" << ConfigSection::shardDatabase() << "] " << msg;
         return false;
+    };
+
+    Config const& config {app_.config()};
+    Section const& section {config.section(ConfigSection::shardDatabase())};
+    if (section.empty())
+        return fail("missing configuration");
+
+    {
+        // Node and shard stores must use same earliest ledger sequence
+        std::uint32_t seq;
+        if (get_if_exists<std::uint32_t>(
+                config.section(ConfigSection::nodeDatabase()),
+                "earliest_seq",
+                seq))
+        {
+            std::uint32_t seq2;
+            if (get_if_exists<std::uint32_t>(section, "earliest_seq", seq2) &&
+                seq != seq2)
+            {
+                return fail("and [" + ConfigSection::shardDatabase() +
+                    "] both define 'earliest_seq'");
+            }
+        }
+    }
+
+    if (!get_if_exists<boost::filesystem::path>(section, "path", dir_))
+        return fail("'path' missing");
+
+    if (boost::filesystem::exists(dir_))
+    {
+        if (!boost::filesystem::is_directory(dir_))
+            return fail("'path' must be a directory");
+    }
+    else
+        boost::filesystem::create_directories(dir_);
+
+    if (!get_if_exists<std::uint64_t>(section, "max_size_gb", maxDiskSpace_))
```

Comment:
I'm pretty sure the existing comparisons with this value are in terms of bytes, but this value is gb. The old code converted to bytes before passing to this class, so we should probably do it here after reading and checking overflow.

Reply:
Thanks! 👍